include/openenclave/internal/posix/resolver.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _OE_POSIX_RESOLVER_H
+#define _OE_POSIX_RESOLVER_H
+
+#include <openenclave/bits/types.h>
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/netdb.h>
+
+OE_EXTERNC_BEGIN
+
+typedef struct _oe_resolver oe_resolver_t;
+
+#define OE_RESOLVER_ENCLAVE_LOCAL 0
+#define OE_RESOLVER_ENCLAVE_DNS 1
+#define OE_RESOLVER_HOST 2
+
+typedef struct _oe_resolver_ops
+{
+    int (*getaddrinfo)(
+        oe_resolver_t* dev,
+        const char* node,
+        const char* service,
+        const struct oe_addrinfo* hints,
+        struct oe_addrinfo** res);
+
+    ssize_t (*getnameinfo)(
+        oe_resolver_t* dev,
+        const struct oe_sockaddr* sa,
+        oe_socklen_t salen,
+        char* host,
+        oe_socklen_t hostlen,
+        char* serv,
+        oe_socklen_t servlen,
+        int flags);
+
+    int (*shutdown)(oe_resolver_t* dev);
+
+} oe_resolver_ops_t;
+
+typedef struct _oe_resolver
+{
+    /* The type of the device. */
+    int type;
+
```

Comment:
Consider making this an enum type like `oe_device_type_t` ... the comment should probably also be clarified to refer to the type of resolver rather than type of device, which has a distinct meaning in this set of headers.

Reply:
Added **oe_resolver_type_t**.